examples/p4est_2d_dgsem/elixir_advection_nonconforming_flag.jl

```diff
@@ -28,7 +28,8 @@ mesh = P4estMesh(trees_per_dimension, polydeg=3,
 
 # Refine bottom left quadrant of each tree to level 4
 function refine_fn(p4est, which_tree, quadrant)
-  if quadrant.x == 0 && quadrant.y == 0 && quadrant.level < 4
+  quadrant_obj = unsafe_load(quadrant)
+  if quadrant_obj.x == 0 && quadrant_obj.y == 0 && quadrant_obj.level < 4
```

Comment:
Not sure if we have discussed this yet... In case we only need to access this data locally, wouldn't it make sense to instead of loading the entire quadrant data to only copy the relevant data fields? I assume `unsafe_load` is allocating, while `x`, `y`, and `z` are just POD values and would thus live on the stack.

We should at least once check the performance impact this has. But In the spirit of make it work, make it nice, make it fast, I'd be ok if we create an issue for this general issue and move ahead here.

Reply:
I added it to the list in https://github.com/trixi-framework/P4est.jl/issues/54 for @JoshuaLampert

Reply:
Great, thanks!